packages/engine/tests/simulations/experiment.rs

```diff
@@ -0,0 +1,292 @@
+            .map(|output_dir_capture| {
+                let output_dir = Path::new(&output_dir_capture[1]);
+                let output_dir_base = output_dir.parent().unwrap().to_path_buf();
+                let simulation_number = output_dir
+                    .strip_prefix(&output_dir_base)
+                    .unwrap()
+                    .to_string_lossy()
+                    .parse::<u64>()
+                    .expect("Unable to parse simulation run ID as integer");
+                (output_dir_base, simulation_number)
+            })
+            .collect::<Vec<_>>();
+        outputs.sort_unstable();
+        outputs
+            .into_iter()
+            .map(|(output_dir_base, simulation_number)| {
+                let output_dir = output_dir_base.join(simulation_number.to_string());
+                let json_state = parse_file(Path::new(&output_dir).join("json_state.json"))
+                    .wrap_err("Could not read JSON state")?;
+                let globals = parse_file(Path::new(&output_dir).join("globals.json"))
+                    .wrap_err("Could not read globals")?;
+                let analysis_outputs =
+                    parse_file(Path::new(&output_dir).join("analysis_outputs.json"))
+                        .wrap_err("Could not read analysis outputs`")?;
+
+                let _ = fs::remove_dir_all(&output_dir);
+
+                Ok((json_state, globals, analysis_outputs))
+            })
+            .collect()
+    }
+}
+
+fn parse_file<T: DeserializeOwned, P: AsRef<Path>>(path: P) -> Result<T> {
+    let path = path.as_ref();
+    serde_json::from_reader(BufReader::new(
+        File::open(path).wrap_err_lazy(|| format!("Could not open file {path:?}"))?,
+    ))
+    .wrap_err_lazy(|| format!("Could not parse {path:?}"))
+}
+
+pub fn read_config<P: AsRef<Path>>(path: P) -> Result<Vec<(Experiment, Vec<ExpectedOutput>)>> {
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    pub enum ConfigValue {
+        #[serde(rename_all = "kebab-case")]
+        Simple {
+            experiment: String,
+            expected_outputs: Vec<ExpectedOutput>,
+        },
+        #[serde(rename_all = "kebab-case")]
+        SingleRun {
+            steps: u64,
+            expected_output: ExpectedOutput,
+        },
+    }
+
+    Ok(parse_file::<Vec<ConfigValue>, P>(path)
+        .wrap_err("Could not read integration test configuration")?
+        .into_iter()
+        .map(|config_value| match config_value {
+            ConfigValue::Simple {
+                experiment,
+                expected_outputs,
+            } => (Experiment::Simple { experiment }, expected_outputs),
+            ConfigValue::SingleRun {
+                steps,
+                expected_output,
+            } => (Experiment::SingleRun { steps }, vec![expected_output]),
+        })
+        .collect())
+}
+
+fn assert_subset_value(subset: &Value, superset: &Value, path: String) -> Result<()> {
```

Comment:
This could have a docstring referring to the docstring of `assert_subset_of` below.

Reply:
```suggestion
/// Implementation for [`ExpectedOutput::assert_subset_of`]
fn assert_subset_value(subset: &Value, superset: &Value, path: String) -> Result<()> {
```